### ISSUE TITLE
Can't upgrade nvm

### ISSUE THREAD

**Issue by alextes** [lukechilds/zsh-nvm#44]
Tracked the problem down to the `_zsh_nvm_latest_release_tag` function.
Command is 
```sh
echo $(cd "$NVM_DIR" && git fetch --quiet origin && git describe --abbrev=0 --tags --match "v[0-9]*" origin)
```
Specifically:
```sh
git describe --abbrev=0 --tags --match "v[0-9]*" origin
```
is coming back as:
```
fatal: Not a valid object name origin
```

`git ls-remote --tags origin` lists the tags fine.

**Comment by alextes** [comment]
Solved it by using `git fetch --unshallow`. Seems it was somehow too shallow to use origin which is weird, since I checked the remotes and origin was there.

**Comment by lukechilds** [comment]
Hey, thanks for the detailed feedback.

I'm following the official git install/upgrade instructions from `nvm`:

https://github.com/creationix/nvm#manual-upgrade

Looks like it requires git v1.7.10+. What git version are you running?

**Comment by runofthemill** [comment]
Had the same issue, and the fix in #46 worked for me.

```
$ git --version
git version 2.17.0
```

🤙🏻

**Comment by Enochenti** [comment]
Used the `git fetch --unshallow` ,`fatal: --unshallow on a complete repository does not make sense` given,maybe it`s not the same issue?

**Comment by cocoonkid** [comment]
Having the same Issue here.   Neat and fresh installed Mojave.

```
$nvm upgrade
Installed version is v0.33.2
Checking latest version of nvm...
fatal: Not a valid object name origin
Updating to ...
fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
```


```
.nvm on  tags/v0.33.2 via ⬢ v11.4.0
  git fetch --unshallow
fatal: --unshallow on a complete repository does not make sense
```

git version 2.20.0

**Comment by alextes** [comment]
Hm, so the reason why this seems to be happening. `zsh-nvm` offers to update `nvm`. It does this by opening the nvm dir, which is a git repo, and using `git-describe` to figure out the most recent tag as the `nvm` people use git tags for releases. `git-describe` takes a reference to a git object, which points at a commit, it checks if the commit has a tag and if not checks its parent. Give it `origin` which I think becomes `origin/master` and you should be walking from the most recent commit backward.

The problem I think is that the `nvm` people set the following git config:
```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "origin"]
	url = https://github.com/creationix/nvm.git
	fetch = +refs/tags/v0.33.6:refs/tags/v0.33.6
```

Now given this I can imagine why git fails to pull in the origin/master tree, `git-fetch` is restricted to only pull in tag 0.33.6 it seems. What I don't understand is how `zsh-nvm` ever managed to upgrade nvm in the first place 😅 .

**Comment by alextes** [comment]
Closed by #46 .
